girder/models/item.py

```diff
@@ -438,6 +443,15 @@ def stream():
                 yield json.dumps(doc['meta'], default=str)
             yield (os.path.join(path, metadataFile), stream)
 
+    def _mimeFilter(self, file, mimeFilter):
+        """
+        Returns whether or not the given file should be passed through the given
+        MIME filter. If no MIME filter is specified, all files are allowed.
+        """
+        if not mimeFilter:
```

Comment:
Request filters could be built programmatically, and having the edge case be "an empty whitelist really means include everything" is a little weird. You could also want to use an empty filter with the metadata option turned on to download only the metadata as a ZIP. IMO, it should be that an empty list filters all files, but it's your call.